src/ansys/dpf/core/data_sources.py

```diff
@@ -69,13 +77,20 @@ class DataSources:
     Initialize a model from a result path.
 
     >>> from ansys.dpf import core as dpf
-    >>> my_data_sources = dpf.DataSources('file.rst')
+    >>> # Create the DataSources object with a main file path
+    >>> my_data_sources = dpf.DataSources(result_path='file.rst')
+    >>> # Get the path to the main result file
     >>> my_data_sources.result_files
     ['file.rst']
 
     """
 
-    def __init__(self, result_path=None, data_sources=None, server=None):
+    def __init__(
+        self,
+        result_path: Optional[str, os.PathLike] = None,
+        data_sources: Optional[dpf.DataSources, int, data_sources_pb2.DataSources] = None,
+        server: Optional[type[server_types.BaseServer]] = None,
```

Comment:
@luisaFelixSalles here this would mean the `server` argument must be a class. Instead, it should be an instance of one of the subclasses of `server_types.BaseServer`. I do not think this is correct.